src/mixins/java/org/spongepowered/common/mixin/core/server/level/PlayerSpawnFinderMixin.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.mixin.core.server.level;
+
+import net.minecraft.CrashReport;
+import net.minecraft.CrashReportCategory;
+import net.minecraft.ReportedException;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.SectionPos;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.level.PlayerSpawnFinder;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.server.level.TicketType;
+import net.minecraft.world.level.ChunkPos;
+import net.minecraft.world.level.CollisionGetter;
+import net.minecraft.world.phys.Vec3;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Shadow;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Supplier;
+
+import javax.annotation.Nullable;
+
+@Mixin(PlayerSpawnFinder.class)
+public abstract class PlayerSpawnFinderMixin {
+
+    @Shadow @Final private CompletableFuture<Vec3> finishedFuture;
+    @Shadow @Final private ServerLevel level;
+    @Shadow @Final private BlockPos spawnSuggestion;
+    @Shadow @Final private int radius;
+    @Shadow @Final private int candidateCount;
+    @Shadow private int nextCandidateIndex;
+    @Shadow @Final private int offset;
+    @Shadow @Final private int coprime;
+
+    @Shadow
+    private static Vec3 fixupSpawnHeight(CollisionGetter $$0, BlockPos $$1) {
+        throw new IllegalStateException();
+    }
```

Comment:
We usually just squash these like this so the LOC's stay small.
```suggestion
    @Shadow private static Vec3 fixupSpawnHeight(CollisionGetter $$0, BlockPos $$1) { throw null; }
```